CIPs/0003.md

```diff
@@ -0,0 +1,134 @@
+# CIP [0003]: Identity Claims and Metadata
+
+- Date: 2019-09-19
+- Author: @timmoreton
+- Status: WIP
+
+## Overview
+
+This CIP describes a general mechanism that allows principals on the Celo platform to claim real-world identities and maintain rich, unverified metadata. Metadata may be public, or optionally, encrypted and accessible only to the Celo Foundation. Third parties can verify these identity claims, and use the associated unverified metadata as appropriate. 
+
+There are a number of principals that provide resources to or deploy services on the Celo platform that may wish to assert identities or provide metadata. These include, but are not limited to:
+* Validators
+* Validator groups
+* Smart contract owners
+* DApp developers
+* Creators of Governance proposals
+* Oracles 
+
+Use cases include:
+* Tools want to present public metadata supplied by a validator or validator group as part of a list of candidate groups, or a list of current elected validators.
+* Governance Explorer UIs may want to present public metadata about the creators of governance proposals
+* The Celo Foundation receives notice of a security vulnerability and wants to contact elected validators to facilitate them to make a decision on applying a patch.
+* A DApp makes a request to the Celo Wallet for account information or to sign a transaction. The Celo Wallet should provide information about the DApp to allow the user to make a decision whether to accept the transaction or not.   
+
+## Goals
+
+- Integrate with the existing Attestations structure, rather than maintain separate identifer fields in other contracts, so that the same identity claims and metadata can be linked from any of a principals' activities  
+- Allow principals to make an extensible range of identity claims, and support clients verifying the veracity of these 
+- Allow principals to supply unverified extensible, structured metadata off-chain
+- Allow tools to choose what metadata they require and how they present it
+- Support both public (unencrypted) metadata and metadata accessible only to the Celo Foundation
+
+## Proposed Solution
+
+### Claims
+
+The Attestations contract is extended as follows:
```

Comment:
Actually, what do you think about hosting claims entirely off-chain? Thinking more about it, there seems to be no reason to really put this in a smart contract other than being a convenient spot, and the storage costs do not seem worth it?

Reply:
I think it could support future on-chain verification of these claims. Maybe phone numbers are just a special case where we've been able to provide an on-chain verification.

Reply:
I think the attestation of those claims can be done with the existing attestations struct, and the claims being hosted off-chain shouldn't impact this?

Reply:
Phone numbers are special because they are not easily verifiable in a "permissionless" way which is why we delegate that task to the protocol. Twitter/DNS can be verified by anyone easily.